In [1]:
import csv 
import json 
import pandas as pd
pd.set_option("display.max_columns", None)


In [2]:
#sacar unnamed 
df = pd.read_csv('hr_raw_data.csv', index_col=0)

In [19]:
df

age attrition businesstravel    dailyrate                department  \
0     51        No            NaN  2015.722222                       NaN   
1     52        No            NaN  2063.388889                       NaN   
2     42        No  travel_rarely  1984.253968   Research & Development    
3     47        No  travel_rarely  1771.404762                       NaN   
4     46        No            NaN  1582.771346                       NaN   
...   ..       ...            ...          ...                       ...   
1673  43        No            NaN   488.944444                       NaN   
1674  47        No            NaN  1973.984127                       NaN   
1675  29        No  travel_rarely   290.035510                       NaN   
1676  47        No  travel_rarely  1032.487286                       NaN   
1677  32        No            NaN   556.256661                       NaN   

      distancefromhome  education    educationfield  employeecount  \
0                    6          3               NaN              1   
1                    1          4     Life Sciences              1   
2                    4          2  Technical Degree              1   
3                    2          4           Medical              1   
4                    3          3  Technical Degree              1   
...                ...        ...               ...            ...   
1673               -26          3           Medical              1   
1674                26          4               NaN              1   
1675                15          3               NaN              1   
1676                 4          3     Life Sciences              1   
1677                 2          2     Life Sciences              1   

      employeenumber  environmentsatisfaction  gender  hourlyrate  \
0                  1                        1       0         NaN   
1                  2                        3       0         NaN   
2                  3                        3       0         NaN   
3                  4                        1       1         NaN   
4                  5                        1       1         NaN   
...              ...                      ...     ...         ...   
1673             824                        2       1         NaN   
1674            1087                        4       1         NaN   
1675             528                        3       0         NaN   
1676              76                        3       1         NaN   
1677             401                        4       1   69.532083   

      jobinvolvement  joblevel                   jobrole  jobsatisfaction  \
0                  3         5        resEArch DIREcToR                 3   
1                  2         5                  ManAGeR                 3   
2                  3         5                  ManaGER                 4   
3                  3         4        ReseArCH DIrECtOr                 3   
4                  4         4          sAleS EXECUtIve                 1   
...              ...       ...                       ...              ...   
1673               4         1       rESEaRcH SciEnTiST                 3   
1674               3         5                  mANager                 3   
1675               3         1       reSearch sCienTiSt                 4   
1676               2         3   maNufACTURING DIREctOr                 2   
1677               3         2       resEArch scIENTisT                 3   

     maritalstatus monthlyincome monthlyrate  numcompaniesworked over18  \
0              NaN     16280,83$   42330,17$                   7      Y   
1              NaN           NaN   43331,17$                   0    NaN   
2          Married           NaN   41669,33$                   1    NaN   
3          Married     14307,50$   37199,50$                   3      Y   
4         Divorced     12783,92$   33238,20$                   2      Y   
...            ...           ...         ...   

In [ ]:
df["attrition"].isna().sum() #attrition no tiene nulos

0

In [ ]:
df["attrition"].unique().sum() 

'NoYes'

In [28]:
df["businesstravel"].isna().sum()

801

In [30]:
df["businesstravel"].unique()

array([nan, 'travel_rarely', 'travel_frequently', 'non-travel'],
      dtype=object)

In [31]:
df["dailyrate"].isna().sum()

0

In [32]:
df["dailyrate"].unique()

array([2015.72222222, 2063.38888889, 1984.25396825, 1771.4047619 ,
       1582.77134647, 1771.92063492, 1032.48728618,  556.256661  ,
       1712.18253968, 1973.98412698, 2060.70634921, 1382.74603175,
       1362.52380952, 2016.54761905, 1861.37301587, 1985.6984127 ,
       1108.92063492, 1728.38095238,  639.78571429, 2047.5       ,
        412.8015873 , 1077.86507937, 1657.3968254 ,  331.19047619,
       1059.19047619, 1080.75396825,  635.76190476,  487.08730159,
        290.03550974,  476.15079365, 1749.73809524, 1063.31746032,
        557.76190476,  608.93650794, 1119.96031746,  610.17460317,
        685.6984127 , 1441.65873016, 1374.28571429,  689.92857143,
        488.53174603, 1431.23809524, 1964.96031746, 1652.34126984,
        991.81746032, 1187.53968254,  444.26984127, 1758.71428571,
        522.78571429,  380.92063492, 2047.70634921,  238.12698413,
        592.94444444, 1075.28571429,  654.84920635,  242.25396825,
        316.95238095, 1288.65079365,  827.46031746, 1760.98412

In [33]:
df["department"].isna().sum()

1366

In [34]:
df["department"].unique()

array([nan, ' Research & Development ', ' Sales ', ' Human Resources '],
      dtype=object)

In [35]:
df["education"].isna().sum()

0

In [36]:
df["education"].unique()

array([3, 4, 2, 1, 5], dtype=int64)

In [37]:
df["educationfield"].isna().sum()

774

In [38]:
df["educationfield"].unique()

array([nan, 'Life Sciences', 'Technical Degree', 'Medical', 'Other',
       'Marketing', 'Human Resources'], dtype=object)

In [39]:
df["employeecount"].isna().sum()

0

In [40]:
df["employeecount"].unique()

array([1], dtype=int64)

In [41]:
df["employeenumber"].isna().sum()

0

In [42]:
df["employeenumber"].unique()

array([   1,    2,    3, ..., 1612, 1613, 1614], dtype=int64)

In [48]:
df["employeenumber"].sample(50)

1112    1113
55        56
141      142
191      192
848      849
374      375
1130    1131
1117    1118
864      865
539      540
1127    1128
338      339
1336    1337
973      974
970      971
49        50
679      680
1524    1525
267      268
513      514
1376    1377
1231    1232
1299    1300
528      529
106      107
1643    1525
721      722
723      724
1086    1087
962      963
1190    1191
1036    1037
278      279
941      942
59        60
597      598
879      880
289      290
1335    1336
127      128
1209    1210
1302    1303
25        26
392      393
1606    1607
1536    1537
676      677
992      993
718      719
173      174
Name: employeenumber, dtype: int64

In [49]:
df["environmentsatisfaction"].unique()

array([ 1,  3,  4,  2, 42, 37, 35, 25, 27, 31, 39, 21, 15, 14, 33, 19, 12,
       13, 28, 47, 36, 29, 24, 46, 16, 22, 41, 49, 11, 48, 18, 10, 45, 38,
       17, 20, 26, 43], dtype=int64)

In [50]:
df["environmentsatisfaction"].isna().sum()

0

In [69]:
df["environmentsatisfaction"].sample(50)

399     29
186      4
1345     4
393      3
401      3
452      4
192      1
894      3
745      4
204      2
1126    35
1144     1
435      3
1529     2
874     17
625      1
656      2
665      2
842      3
115     39
936      4
1205     1
1658     2
283      4
1104     3
0        1
1656     2
241      4
1340     3
193      2
406      4
861      1
733     42
188      4
627     41
1470     2
821      3
1587     4
1609     3
1663     2
79       4
501      4
1108     1
523     12
1661     3
68      25
48       4
640      1
1516     1
1023     4
Name: environmentsatisfaction, dtype: int64

In [70]:
df["hourlyrate"].unique()

array([         nan,  69.53208262, 172.84325397, 216.04761905,
        79.97321429, 129.06091077, 246.74801587, 207.17460317,
        60.8859127 ,  36.25443872, 139.99503968,  86.24107143,
       197.84641831,  61.06646825, 245.62003968, 123.97718254,
       148.44246032, 255.96329365, 134.41071429,  71.92559524,
       113.78869048, 175.10019841, 247.49007937, 220.83234127,
        91.34821429,  37.77480159, 198.9593254 , 124.05456349,
       247.58035714, 108.48809524,  43.17857143,  68.46924603,
        32.48710317,  55.26289683,  83.46825397,  79.26388889,
        72.28670635,  45.96428571,  31.66170635,  65.19345238,
        52.24503968, 143.9156746 , 113.40178571, 143.19345238,
       139.86607143,  70.57142857,  97.04861111,  45.02281746,
       223.47619048, 134.9781746 ,  74.40178571,  67.14087302,
        58.06150794,  35.05357143,  73.18948413, 164.33134921,
        43.53968254,  54.79861111, 252.59722222,  83.81646825,
        46.71230159,  33.00297619,  70.77777778,  69.23

In [71]:
df["hourlyrate"].isna().sum()

1267

In [72]:
df["jobinvolvement"].unique()

array([3, 2, 4, 1], dtype=int64)

In [73]:
df["jobinvolvement"].isna().sum()

0

In [74]:
df["joblevel"].unique()

array([5, 4, 3, 2, 1], dtype=int64)

In [75]:
df["joblevel"].isna().sum()

0

In [76]:
df["jobsatisfaction"].unique()

array([3, 4, 1, 2], dtype=int64)

In [77]:
df["jobsatisfaction"].isna().sum()

0

In [78]:
df["maritalstatus"].unique()

array([nan, 'Married', 'Divorced', 'Single', 'divorced', 'Marreid'],
      dtype=object)

In [79]:
df["maritalstatus"].isna().sum()

675

In [98]:
df["percentsalaryhike"].unique()

array([13, 14, 11, 19, 12, 25, 16, 17, 22, 23, 20, 15, 21, 24, 18],
      dtype=int64)

In [99]:
df["percentsalaryhike"].isna().sum()

0

In [100]:
df["performancerating"].unique()

array(['3,0', '4,0', nan], dtype=object)

In [101]:
df["performancerating"].isna().sum()

200

In [102]:
df["relationshipsatisfaction"].unique()

array([3, 1, 4, 2], dtype=int64)

In [103]:
df["relationshipsatisfaction"].isna().sum()

0

In [106]:
df["stockoptionlevel"].unique()

array([0, 1, 2, 3], dtype=int64)

In [107]:
df["stockoptionlevel"].isna().sum()

0

In [108]:
df["totalworkingyears"].unique()

array([nan, '34,0', '22,0', '28,0', '20,0', '21,0', '33,0', '40,0',
       '18,0', '25,0', '15,0', '17,0', '26,0', '16,0', '24,0', '14,0',
       '23,0', '27,0', '19,0', '11,0', '38,0', '37,0', '13,0', '12,0',
       '29,0', '10,0', '36,0', '35,0', '9,0', '31,0', '32,0', '8,0',
       '7,0', '30,0', '6,0', '5,0', '4,0', '3,0', '2,0', '1,0', '0,0'],
      dtype=object)

In [109]:
df["totalworkingyears"].isna().sum()

549

In [110]:
df["trainingtimeslastyear"].unique()

array([5, 3, 2, 0, 1, 4, 6], dtype=int64)

In [111]:
df["trainingtimeslastyear"].isna().sum()

0

In [112]:
df["worklifebalance"].unique()

array(['3,0', nan, '2,0', '4,0', '1,0'], dtype=object)

In [113]:
df["worklifebalance"].isna().sum()

114

In [114]:
df["yearsatcompany"].unique()

array([20, 33, 22, 19, 21, 18, 24, 31, 26, 16, 23, 15, 17, 32, 14, 13, 25,
       12, 11, 37, 40, 36, 27, 29, 10,  9, 30,  8,  7, 34,  6,  5,  4,  2,
        3,  1,  0], dtype=int64)

In [115]:
df["yearsatcompany"].isna().sum()

0

In [116]:
df["yearsincurrentrole"].unique()

array([nan, '13,0', '12,0', '11,0', '7,0', '6,0', '4,0', '3,0', '2,0',
       '1,0', '0,0'], dtype=object)

In [117]:
df["yearsincurrentrole"].isna().sum()

1643

In [118]:
df["yearssincelastpromotion"].unique()

array([15, 11,  5,  2,  4,  7,  0,  1, 13, 14,  8, 12,  3,  6, 10,  9],
      dtype=int64)

In [119]:
df["yearssincelastpromotion"].isna().sum()

0

In [120]:
df["yearswithcurrmanager"].unique()

array([15,  9,  6,  8,  7, 11, 10, 12,  4,  0,  5, 17,  2, 14,  1, 13,  3,
       16], dtype=int64)

In [121]:
df["yearswithcurrmanager"].isna().sum()

0

In [88]:
min(df["monthlyincome"])

TypeError: '<' not supported between instances of 'float' and 'str'

In [80]:
df["monthlyincome"].unique()

array(['16280,83$', nan, '14307,50$', '12783,92$', '14311,67$',
       '8339,32$', '13829,17$', '15943,72$', '16644,17$', '15034,17$',
       '16038,33$', '4492,84$', '8956,67$', '5167,50$', '3334,17$',
       '8705,83$', '2675,00$', '8555,00$', '8729,17$', '3934,17$',
       '3845,83$', '14132,50$', '8588,33$', '4505,00$', '4918,33$',
       '4928,33$', '5538,33$', '11100,00$', '5572,50$', '11560,00$',
       '15870,83$', '13345,83$', '8010,83$', '3588,33$', '14205,00$',
       '4222,50$', '3076,67$', '2342,59$', '4789,17$', '8685,00$',
       '5289,17$', '1956,67$', '10408,33$', '6683,33$', '7452,50$',
       '16060,00$', '4647,50$', '2242,50$', '6169,17$', '7352,50$',
       '2982,50$', '14815,83$', '15177,50$', '15991,67$', '15989,17$',
       '8340,00$', '6656,67$', '5902,50$', '3935,83$', '2839,17$',
       '2440,83$', '10025,83$', '12855,83$', '5135,00$', '4271,67$',
       '8015,83$', '4175,00$', '15860,83$', '8666,67$', '2327,50$',
       '4728,33$', '7010,00$', '2790,00$', '4

In [89]:
df["monthlyrate"].isna().sum()

0

In [90]:
df["monthlyrate"].unique()

array(['42330,17$', '43331,17$', '41669,33$', '37199,50$', '33238,20$',
       '37210,33$', '21682,23$', '11681,39$', '35955,83$', '41453,67$',
       '43274,83$', '29037,67$', '28613,00$', '42347,50$', '39088,83$',
       '41699,67$', '23287,33$', '36296,00$', '13435,50$', '42997,50$',
       '8668,83$', '22635,17$', '34805,33$', '6955,00$', '22243,00$',
       '22695,83$', '13351,00$', '10228,83$', '6090,75$', '9999,17$',
       '36744,50$', '22329,67$', '11713,00$', '12787,67$', '23519,17$',
       '12813,67$', '14399,67$', '30274,83$', '28860,00$', '14488,50$',
       '10259,17$', '30056,00$', '41264,17$', '34699,17$', '20828,17$',
       '24938,33$', '9329,67$', '36933,00$', '10978,50$', '7999,33$',
       '43001,83$', '5000,67$', '12451,83$', '22581,00$', '13751,83$',
       '5087,33$', '6656,00$', '27061,67$', '17376,67$', '36980,67$',
       '19376,50$', '41756,00$', '12083,50$', '5830,50$', '16039,83$',
       '19116,50$', '7754,50$', '38521,17$', '39461,50$', '29416,83$',
   

In [91]:
df["numcompaniesworked"].unique()

array([7, 0, 1, 3, 2, 4, 8, 9, 5, 6], dtype=int64)

In [93]:
df["numcompaniesworked"].isna().sum()

0

In [94]:
df["over18"].unique()

array(['Y', nan], dtype=object)

In [95]:
df["over18"].isna().sum()

938

In [96]:
df["overtime"].isna().sum()

696

In [97]:
df["overtime"].unique()

array(['No', nan, 'Yes'], dtype=object)

In [104]:
df["standardhours"].isna().sum()

351

In [105]:
df["standardhours"].unique()

array(['Full Time', nan, 'Part Time'], dtype=object)

In [122]:
df["sameasmonthlyincome"].isna().sum()

489

In [123]:
df["sameasmonthlyincome"].unique()

array(['16280,83$', nan, '14307,50$', '12783,92$', '14311,67$',
       '8339,32$', '13829,17$', '15943,72$', '16644,17$', '15034,17$',
       '16038,33$', '4492,84$', '8956,67$', '5167,50$', '3334,17$',
       '8705,83$', '2675,00$', '8555,00$', '8729,17$', '3934,17$',
       '3845,83$', '14132,50$', '8588,33$', '4505,00$', '4918,33$',
       '4928,33$', '5538,33$', '11100,00$', '5572,50$', '11560,00$',
       '15870,83$', '13345,83$', '8010,83$', '3588,33$', '14205,00$',
       '4222,50$', '3076,67$', '2342,59$', '4789,17$', '8685,00$',
       '5289,17$', '1956,67$', '10408,33$', '6683,33$', '7452,50$',
       '16060,00$', '4647,50$', '2242,50$', '6169,17$', '7352,50$',
       '2982,50$', '14815,83$', '15177,50$', '15991,67$', '15989,17$',
       '8340,00$', '6656,67$', '5902,50$', '3935,83$', '2839,17$',
       '2440,83$', '10025,83$', '12855,83$', '5135,00$', '4271,67$',
       '8015,83$', '4175,00$', '15860,83$', '8666,67$', '2327,50$',
       '4728,33$', '7010,00$', '2790,00$', '4

In [124]:
df["datebirth"].isna().sum()

0

In [125]:
df["datebirth"].unique()

array([1972, 1971, 1981, 1976, 1977, 1975, 1964, 1982, 1967, 1985, 1968,
       1983, 1965, 1988, 1978, 1990, 1987, 1989, 1970, 1980, 1963, 1991,
       1986, 1974, 1984, 1973, 1979, 1993, 1994, 1992, 1969, 1966, 1996,
       1995, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005],
      dtype=int64)

In [126]:
df["salary"].isna().sum()

285

In [127]:
df["salary"].unique()

array(['195370,00$', '199990,00$', '192320,00$', '171690,00$', nan,
       '100071,84$', '53914,11$', '165950,00$', '199730,00$',
       '134020,00$', '132060,00$', '195450,00$', '180410,00$',
       '192460,00$', '107480,00$', '167520,00$', '62010,00$',
       '198450,00$', '40010,00$', '104470,00$', '191324,62$',
       '160640,00$', '32100,00$', '102660,00$', '104750,00$', '61620,00$',
       '47210,00$', '153407,07$', '28111,13$', '46150,00$', '169590,00$',
       '54060,00$', '59020,00$', '108550,00$', '59140,00$', '66460,00$',
       '139730,00$', '66870,00$', '47350,00$', '138720,00$', '190450,00$',
       '96130,00$', '115100,00$', '170460,00$', '36920,00$', '198470,00$',
       '23080,00$', '57470,00$', '104220,00$', '63470,00$', '23480,00$',
       '30720,00$', '124900,00$', '80200,00$', '170680,00$', '89430,00$',
       '192720,00$', '55770,00$', '26910,00$', '74030,00$', '88230,00$',
       '35790,00$', '177790,00$', '182130,00$', '135770,00$',
       '191900,00$', '171230,

In [128]:
df["roledepartament"].isna().sum()

1366

In [129]:
df["roledepartament"].unique()

array([nan, ' ManaGER  -  Research & Development ',
       ' MANAger  -  Research & Development ',
       ' heaLtHcArE repResENtATiVe  -  Research & Development ',
       ' saLEs exEcutIVe  -  Sales ', ' Sales ExECuTIVe  -  Sales ',
       ' heALthCArE RePResEntaTIVe  -  Research & Development ',
       ' lABORAtOry tECHnIcIan  -  Research & Development ',
       ' ManufactUrINg DiReCTOr  -  Research & Development ',
       ' SaleS eXEcUtIVE  -  Sales ',
       ' HEaLthcaRe rEprEsEnTAtiVe  -  Research & Development ',
       ' rESeArCH sCientIST  -  Research & Development ',
       ' healTHCAre rePREseNtATiVE  -  Research & Development ',
       ' mANaGer  -  Research & Development ',
       ' HEAltHCaRe REPreseNTatIve  -  Research & Development ',
       ' SaLeS eXECUTivE  -  Sales ',
       ' reSeARCH SCIEnTiST  -  Research & Development ',
       ' ReSearch DiRecTOR  -  Research & Development ',
       ' resEaRCh SciEnTist  -  Research & Development ',
       ' hUmaN REsoUrCes  -  H

In [130]:
df["numberchildren"].unique()

array([nan])

In [131]:
df["numberchildren"].isna().sum()

1678

In [132]:
df["remotework"].isna().sum()

0

In [133]:
df["remotework"].unique()

array(['Yes', '1', 'False', '0', 'True'], dtype=object)

In [ ]:
df.columns 

Index(['age', 'attrition', 'businesstravel', 'dailyrate', 'department',
       'distancefromhome', 'education', 'educationfield', 'employeecount',
       'employeenumber', 'environmentsatisfaction', 'gender', 'hourlyrate',
       'jobinvolvement', 'joblevel', 'jobrole', 'jobsatisfaction',
       'maritalstatus', 'monthlyincome', 'monthlyrate', 'numcompaniesworked',
       'over18', 'overtime', 'percentsalaryhike', 'performancerating',
       'relationshipsatisfaction', 'standardhours', 'stockoptionlevel',
       'totalworkingyears', 'trainingtimeslastyear', 'worklifebalance',
       'yearsatcompany', 'yearsincurrentrole', 'yearssincelastpromotion',
       'yearswithcurrmanager', 'sameasmonthlyincome', 'datebirth', 'salary',
       'roledepartament', 'numberchildren', 'remotework'],
      dtype='object')

In [ ]:
#transformar el 0 y 1 por estereotipos de género como masculino y femenino con asignación directa, la empresa debe
#replantearse el non-binary
df['gender'] = df['gender'].replace(0, 'M')

df['gender'] = df['gender'].replace(1, 'F')
print(df['gender'])


0       M
1       M
2       M
3       F
4       F
       ..
1673    F
1674    F
1675    M
1676    F
1677    F
Name: gender, Length: 1678, dtype: object


In [ ]:
#Algunas columnas, como la columna DailyRate son columnas que incluyen valores numéricos decimales. 
# Pero en el DataFrame aparece como columna de tipo string. Deberás hacer los cambios necesarios para 
# convertirla en columna de tipo numérica.
df['dailyrate'].dtype #aquí aparece como float
[type('dailyrate')] # da str

df['dailyrate'] #emo sido engañadas, está correcto 

0       2015.722222
1       2063.388889
2       1984.253968
3       1771.404762
4       1582.771346
           ...     
1673     488.944444
1674    1973.984127
1675     290.035510
1676    1032.487286
1677     556.256661
Name: dailyrate, Length: 1678, dtype: float64

In [ ]:
df.info()


<class 'pandas.core.frame.DataFrame'>
Index: 1678 entries, 0 to 1677
Data columns (total 41 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       1678 non-null   object 
 1   attrition                 1678 non-null   object 
 2   businesstravel            877 non-null    object 
 3   dailyrate                 1678 non-null   float64
 4   department                312 non-null    object 
 5   distancefromhome          1678 non-null   int64  
 6   education                 1678 non-null   int64  
 7   educationfield            904 non-null    object 
 8   employeecount             1678 non-null   int64  
 9   employeenumber            1678 non-null   int64  
 10  environmentsatisfaction   1678 non-null   int64  
 11  gender                    1678 non-null   object 
 12  hourlyrate                411 non-null    float64
 13  jobinvolvement            1678 non-null   int64  
 14  joblevel     


Index: 1678 entries, 0 to 1677
Data columns (total 41 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       1678 non-null   object 
 1   attrition                 1678 non-null   object 
 2   businesstravel            877 non-null    object 
 3   dailyrate                 1678 non-null   float64
 4   department                312 non-null    object 
 5   distancefromhome          1678 non-null   int64  
 6   education                 1678 non-null   int64  
 7   educationfield            904 non-null    object 
 8   employeecount             1678 non-null   int64  
 9   employeenumber            1678 non-null   int64  
 10  environmentsatisfaction   1678 non-null   int64  
 11  gender                    1678 non-null   object 
 12  hourlyrate                411 non-null    float64
 13  jobinvolvement            1678 non-null   int64  
 14  joblevel                  1678 non-null   int64  
 15  jobrole                   1678 non-null   object 
 16  jobsatisfaction           1678 non-null   int64  
 17  maritalstatus             1003 non-null   object 
 18  monthlyincome             1189 non-null   object 
 19  monthlyrate               1678 non-null   object 
 20  numcompaniesworked        1678 non-null   int64  
 21  over18                    740 non-null    object 
 22  overtime                  982 non-null    object 
 23  percentsalaryhike         1678 non-null   int64  
 24  performancerating         1478 non-null   object 
 25  relationshipsatisfaction  1678 non-null   int64  
 26  standardhours             1327 non-null   object 
 27  stockoptionlevel          1678 non-null   int64  
 28  totalworkingyears         1129 non-null   object 
 29  trainingtimeslastyear     1678 non-null   int64  
 30  worklifebalance           1564 non-null   object 
 31  yearsatcompany            1678 non-null   int64  
 32  yearsincurrentrole        35 non-null     object 
 33  yearssincelastpromotion   1678 non-null   int64  
 34  yearswithcurrmanager      1678 non-null   int64  
 35  sameasmonthlyincome       1189 non-null   object 
 36  datebirth                 1678 non-null   int64  
 37  salary                    1393 non-null   object 
 38  roledepartament           312 non-null    object 
 39  numberchildren            0 non-null      float64
 40  remotework                1678 non-null   object 
dtypes: float64(3), int64(17), object(21)
memory usage: 550.6+ KB

In [ ]:
from IPython.display import display

# Solo mostramos la columna 'gender' con scroll
display(df[['age']].style.set_sticky().set_table_attributes('style="height:300px; overflow-y:scroll; display:block"'))




In [ ]:
import re

In [ ]:
df_filtrado = df[df['age'].str.match(r'^.{3,}$')]


In [ ]:
df_filtrado #nos da los errores

age attrition businesstravel    dailyrate  \
31     forty-seven        No            NaN  1032.487286   
79     fifty-eight        No  travel_rarely  1431.238095   
216     thirty-six        No  travel_rarely  1032.487286   
357     fifty-five       Yes            NaN   556.256661   
411      fifty-two       Yes            NaN  1032.487286   
426     thirty-one        No            NaN  1113.563492   
593         thirty        No  travel_rarely   556.256661   
1198    twenty-six       Yes            NaN   245.246032   
1226  thirty-seven       Yes  travel_rarely   215.634921   
1293    thirty-two        No  travel_rarely   433.023810   
1325   twenty-four        No            NaN   277.952381   
1584    thirty-two        No  travel_rarely  1151.325397   
1662   twenty-four        No            NaN   277.952381   

                    department  distancefromhome  education educationfield  \
31     Research & Development                 25          3        Medical   
79                         NaN               -18          4            NaN   
216    Research & Development                 12          5            NaN   
357                        NaN                 2          1        Medical   
411                     Sales                  5          3            NaN   
426                        NaN                 8          2            NaN   
593                        NaN                 2          3  Life Sciences   
1198                       NaN                21          3            NaN   
1226                       NaN                 2          2            NaN   
1293                       NaN                10          4            NaN   
1325                       NaN                 9          3  Life Sciences   
1584                       NaN                 3          2  Life Sciences   
1662                       NaN                 9          3  Life Sciences   

      employeecount  employeenumber  environmentsatisfaction gender  \
31                1              32                        1      M   
79                1              80                        4      M   
216               1             217                        4      F   
357               1             358                        3      M   
411               1             412                        2      M   
426               1             427                        1      M   
593               1             594                        3      F   
1198              1            1199                        1      M   
1226              1            1227                        4      M   
1293              1            1294                        3      M   
1325              1            1326                        3      M   
1584              1            1585                        3      F   
1662              1            1326                        3      M   

      hourlyrate  jobinvolvement  joblevel                      jobrole  \
31           NaN               3         3   heALthCArE RePResEntaTIVe    
79           NaN               3         4             saLes ExECutiVE    
216   129.060911               2         3      mANUfaCTuRiNG DiRECtOr    
357    69.532083               3         2             SALes EXECutIVe    
411          NaN               3         3             SaLeS eXECutiVE    
426          NaN               3         3             Sales eXEcUTive    
593          NaN               1         2      maNUfaCTURING DIrECToR    
1198   30.655754               3         1       LAboRATory tEChnicIAN    
1226         NaN               2         1       lAbOratOry teChnIcIan    
1293   54.127976               3         2      maNuFACtURINg DirEcToR    
1325         NaN               2         1       labORAToRy TeCHNicIAN    
1584         NaN               3         3           researCh dIrEctor    
1662         NaN               2         1       labORAToRy TeCHNicIAN    

      jobsatisfaction maritalstatus

In [ ]:
import numpy as np

In [ ]:
np.where(df['age'] == ["twenty-four"], 24)

ValueError: ('Lengths must match to compare', (1678,), (1,))

In [ ]:
#print(df['age']) #debería de ser Int no object
#df['age']= df['age'].astype(int)
#sin_esclavitud_infantil = [(df['age'] < 16) | (df['age'] < 68)]

def cambiaredad():
    for i in (df['age']) > REGEXPATTEN:

#datebirth investigar restando año actual - datebirth y sustituír a todos.



0       51
1       52
2       42
3       47
4       46
        ..
1673    43
1674    47
1675    29
1676    47
1677    32
Name: age, Length: 1678, dtype: object


ValueError: invalid literal for int() with base 10: 'forty-seven'